web/controllers/page_controller.ex

```diff
@@ -2,14 +2,71 @@ defmodule Pairmotron.PageController do
   use Pairmotron.Web, :controller
 
   alias Pairmotron.User
+  alias Pairmotron.Pair
   alias Pairmotron.Mixer
   alias Pairmotron.Pairer
 
   def index(conn, _params) do
-    {_year, week} = Timex.iso_week(Timex.today)
-    pairs = Repo.all(User.active_users)
+    {year, week} = Timex.iso_week(Timex.today)
+    pairs = fetch_or_gen(year, week)
+    render conn, "index.html", pairs: pairs, year: year, week: week
+  end
+
+  def show(conn, %{"year" => y, "week" => w}) do
+    {year, _} = y |> Integer.parse
+    {week, _} = w |> Integer.parse
+    pairs = fetch_or_gen(year, week)
+    render conn, "index.html", pairs: pairs, year: year, week: week
+  end
+
+  def delete(conn, %{"year" => y, "week" => w}) do
+    {year, _} = y |> Integer.parse
+    {week, _} = w |> Integer.parse
+    fetch_pairs(year, week)
+      |> Enum.map(fn(p) -> Repo.delete! p end)
+    conn
+      |> put_flash(:info, "Repairified")
+      |> redirect(to: page_path(conn, :show, year, week))
+  end
+
+  defp fetch_or_gen(year, week) do
+    case fetch_pairs(year, week) do
+      [] ->
+        generate_pairs(year, week)
+        fetch_pairs(year, week)
+      pairs -> pairs
+    end
+      |> fetch_users_from_pairs
+  end
+
+  defp fetch_pairs(year, week) do
+    Pair
+      |> where(year: ^year, week: ^week)
```

Comment:
Variables in Ecto queries have to be pinned. Ecto will raise an error if you don't

```
** (Ecto.Query.CompileError) variable `year` is not a valid query expression. Variables need to be explicitly interpolated in queries with ^
```